File: registers/data_register/data_register.cc
#include "CAProject/registers/data_register/data_register.hh"

namespace gem5
{

DataRegister::DataRegister(const DataRegisterParams &params) : 
    SimObject(params),
    data(params.data)
{
}

void
DataRegister::setValue(uint16_t val)
{
    data = val;
}

void
DataRegister::increment()
{
    data++;
}

void
DataRegister::clear()
{
    data = 0;
}

uint16_t
DataRegister::getValue()
{
    return data;
}

} // namespace gem5

File: registers/data_register/DataRegister.py
from m5.params import *
from m5.SimObject import SimObject

class DataRegister(SimObject):
    type = 'DataRegister'
    cxx_header = "CAProject/registers/data_register/data_register.hh"
    cxx_class = "gem5::DataRegister"

    data = Param.UInt16(0, "Data stored in the register")
